### PR TITLE
Add Target option for copying frameworks when installing

### PR DESCRIPTION
**Pull request by jsorge**
There's a checkbox in a target's Embed Frameworks step that says "copy only when installing", and it defaults to un-selected. This PR adds the `onlyCopyFilesOnInstall` flag to `Target` which will check that box.

This resolves https://github.com/yonaskolb/XcodeGen/issues/608.

### CODE REVIEW

**Comment by brentleyjones** [comment]
> I'm happy to add some tests around this feature, but I couldn't find the right spot in the project to add them.

A good start is the fixture tests, so we can confirm the actual project output.

**Comment by jsorge** [comment]
@brentleyjones Added a test case to parse the JSON dictionary as well as the flag to a target in our yaml sample. Not sure where to assert that the right changes in the resulting project file occurred though.

File: Sources/ProjectSpec/Target.swift [hunk 1]
```diff
@@ -46,6 +46,7 @@ public struct Target: ProjectTarget {
     public var deploymentTarget: Version?
     public var attributes: [String: Any]
     public var productName: String
+    public var onlyCopyFilesOnInstall: Bool
```
Comment (yonaskolb) [inline Sources/ProjectSpec/Target.swift:49]: Can you write this value in `toJSONValue` down below if the value is not false

**Comment by yonaskolb** [comment]
Thanks @jsorge! It might be useful to also set a `Run script only when installing` on the Carthage build phase. Thoughts? 

In terms of the fixture test, I can see that it led to a diff and that's enough. That's used a simple integration and snapshot test. Uncommitted diffs and compilation errors cause a failure on CI.
If you'd like to add a unit test that would be great too. Then you can assert the exact values.

Could you also add a changelog entry.

**Comment by jsorge** [comment]
@yonaskolb Thanks for the feedback! Addressed everything except the Carthage build phase, since I'm not sure how that works. My mental model is to have this mirror Xcode's checkbox (which is linked in issue #608). How does Carthage interact with the "Embed Frameworks" build phase?

**Comment by yonaskolb** [comment]
@jsorge the generated Carthage build step basically extracts and copies the frameworks in, which is similar. I'm happy to leave this PR focused though. Thanks again!

**Comment by Elland** [comment]
@jsorge thanks!

**Pull request merged**